modules/Core/includes/endpoints/RegisterEndpoint.php

```diff
@@ -0,0 +1,241 @@
+                    ));
+
+                    $api->throwError(14, $api->getLanguage()->get('api', 'unable_to_send_registration_email'));
+                }
+            } else {
+                // PHP mail function
+                $siteemail = Util::getSetting($api->getDb(), 'site_email');
+
+                $to      = $email;
+                $subject = SITE_NAME . ' - ' . $api->getLanguage()->get('emails', 'register_subject');
+
+                $headers = 'From: ' . $siteemail . "\r\n" .
+                'Reply-To: ' . $siteemail . "\r\n" .
+                'X-Mailer: PHP/' . phpversion() . "\r\n" .
+                'MIME-Version: 1.0' . "\r\n" .
+                'Content-type: text/html; charset=UTF-8' . "\r\n";
+
+                $email = array(
+                    'to' => $to,
+                    'subject' => $subject,
+                    'message' => str_replace('[Link]', $link, $html),
+                    'headers' => $headers
+                );
+
+                $sent = Email::send($email, 'php');
+
+                if (isset($sent['error'])) {
+                    // Error, log it
+                    $api->getDb()->insert('email_errors', array(
+                        'type' => 4,
+                        'content' => $sent['error'],
+                        'at' => date('U'),
+                        'user_id' => $user_id
+                    ));
+
+                    $api->throwError(14, $api->getLanguage()->get('api', 'unable_to_send_registration_email'));
+                }
+            }
+
+            $user = new User();
+            HookHandler::executeEvent('registerUser', array(
+                'event' => 'registerUser',
+                'user_id' => $user_id,
+                'username' => Output::getClean($username),
+                'content' => str_replace('{x}', Output::getClean($username), $api->getLanguage()->get('user', 'user_x_has_registered')),
+                'avatar_url' => $user->getAvatar($user_id, null, 128, true),
+                'url' => Util::getSelfURL() . ltrim(URL::build('/profile/' . Output::getClean($username)), '/'),
+                'language' => $api->getLanguage()
+            ));
+
+            $api->returnArray(array('message' => $api->getLanguage()->get('api', 'finish_registration_email')));
+        } else $api->throwError(1, $api->getLanguage()->get('api', 'invalid_api_key'));
+    }
+
+    /**
+     * Inserts new user information to database
+     * 
+     * For internal API use only
+     * 
+     * @see Nameless2API::register()
+     * 
+     * @param string $username The username of the new user to create
+     * @param string $uuid (optional) The Minecraft UUID of the new user
+     * @param string $email The email of the new user
+     * @param string $code The reset token/temp password of the new user
+     * 
+     * @return string JSON Array
+     */
+    private function createUser(Nameless2API $api, $username, $uuid, $email, $code = null) {
```

Comment:
Another type

Reply:
As per our conversation on Discord, leaving this.